### PR TITLE
fix(client): disable env loading for driver adapters

### PR DESCRIPTION
**Pull request by millsp**
[Internal Slack](https://prisma-company.slack.com/archives/C016KUHB1R6/p1695977547807799)

/integration

### CODE REVIEW

File: packages/client/src/runtime/getPrismaClient.ts [hunk 1]
```diff
@@ -343,7 +344,7 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
       }
 
       const loadedEnv = // for node we load the env from files, for edge only via env injections
```
Comment (millsp) [inline packages/client/src/runtime/getPrismaClient.ts:346]: TODO update comment

File: packages/client/src/runtime/getPrismaClient.ts [hunk 2]
```diff
@@ -343,7 +344,7 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
...
-        (NODE_CLIENT && tryLoadEnvs(envPaths, { conflictCheck: 'none' })) || config.injectableEdgeEnv?.()
+        (NODE_CLIENT && adapter && tryLoadEnvs(envPaths, { conflictCheck: 'none' })) || config.injectableEdgeEnv?.()
```
Comment (Jolg42) [inline packages/client/src/runtime/getPrismaClient.ts:347]:
```suggestion
        (NODE_CLIENT && !adapter && tryLoadEnvs(envPaths, { conflictCheck: 'none' })) || config.injectableEdgeEnv?.()
```
We want to not load for driver adapters, so the opposite here

Comment (millsp) [reply packages/client/src/runtime/getPrismaClient.ts:347]: :facepalm:

File: packages/client/tests/e2e/dotenv-loading/_steps.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  setup: async () => {
+    await $`pnpm install`
+    await $`pnpm prisma generate`
+    // await $`pnpm exec prisma db push --force-reset --skip-generate`
+  },
+  test: async () => {
+    // await $`pnpm exec prisma -v`
+    // await $`ts-node src/index.ts`
+    // await $`pnpm exec tsc --noEmit`
+    console.debug('process.env.TEST_ENV_VAR sdjhfsd,ghsdfh', process.env.TEST_ENV_VAR)
```
Comment (millsp) [inline packages/client/tests/e2e/dotenv-loading/_steps.ts:15]: Cleanup later

**Comment by Jolg42** [comment]
ecosystem-tests are looking good https://github.com/prisma/ecosystem-tests/actions/runs/6406267169!

**Comment by Jolg42** [comment]
Note: this did not work as expected in the end, thanks to Jest hiding things from us 🙈

**Pull request merged**